org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ResourceUtils.java

```diff
@@ -35,12 +35,18 @@
  */
 public final class ResourceUtils {
 
+	private static final String ESCAPE_DOLLAR = "\\\\\\$";
+	private static final String DOLLAR = "\\$";
 	public static final String FILE_UNC_PREFIX = "file:////";
 
 	private ResourceUtils() {
 		// No instanciation
 	}
 
+	public static String replaceAll(String replace) {
```

Comment:
It doesn't really relate to Resource manipulations. I'd move it to CompletionUtils.sanitizeCompletion()